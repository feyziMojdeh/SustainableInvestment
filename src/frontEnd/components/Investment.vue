<template>
  <div class="card">
    <Tabs value="0">
      <TabList>
        <Tab value="0">Investment Portfolio</Tab>
        <Tab value="1">Top Growth Companies</Tab>
      </TabList>
      <TabPanels>
        <TabPanel value="0">
          <InvestmentPortfolio :investments="investments"/>
        </TabPanel>
        <TabPanel value="1">
          <TopGrowthCompanies :investments="filteredInvestments" />
        </TabPanel>
      </TabPanels>
    </Tabs>
  </div>
</template>

<script>
import axios from "axios";
import InvestmentPortfolio from "./InvestmentPortfolio.vue";
import Tabs from "primevue/tabs";
import TabList from "primevue/tablist";
import Tab from "primevue/tab";
import TabPanels from "primevue/tabpanels";
import TabPanel from "primevue/tabpanel";
import TopGrowthCompanies from "./TopGrowthCompanies.vue";

export default {
  components: {
    InvestmentPortfolio,
    TopGrowthCompanies,
    Tabs,
    TabList,
    Tab,
    TabPanels,
    TabPanel,
  },
  data() {
    return { investments: [] };
  },
  computed: {
    filteredInvestments() {
      return this.investments
        .filter((company) => company.growth > 21)
        .sort((a, b) => b.growth - a.growth)
        .slice(0, 3);
    },
  },
  methods: {
    async fetchInvestments() {
      try {
        const response = await axios.get("http://localhost:3000/investments");
        this.investments = response.data;
      } catch (error) {
        console.error("Error fetching investments:", error);
      }
    },
  },
  created() {
    this.fetchInvestments();
  },
};
</script>
