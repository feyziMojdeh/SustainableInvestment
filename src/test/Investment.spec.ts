import { shallowMount } from "@vue/test-utils";
import { describe, it, expect, beforeEach, vi } from "vitest";
import axios from "axios";
import Investment from "@/frontEnd/components/Investment.vue";

vi.mock("axios");

describe("Investment", () => {
  const mockInvestments = [
    { id: 1, name: "Company A", growth: 25 },
    { id: 2, name: "Company B", growth: 22 },
    { id: 3, name: "Company C", growth: 30 },
    { id: 4, name: "Company D", growth: 15 },
  ];

  beforeEach(() => {
    axios.get.mockResolvedValue({ data: mockInvestments });
  });

  it("match snapshot", async () => {
    const wrapper = shallowMount(Investment);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("fetches investments on created hook", async () => {
    const wrapper = shallowMount(Investment);

    await wrapper.vm.$nextTick();

    expect(axios.get).toHaveBeenCalledWith("http://localhost:3000/investments");
    expect(wrapper.vm.investments).toEqual(mockInvestments);
  });

  it("computes filteredInvestments correctly", async () => {
    const wrapper = shallowMount(Investment);

    await wrapper.vm.$nextTick();

    expect(wrapper.vm.filteredInvestments).toEqual([
      { id: 3, name: "Company C", growth: 30 },
      { id: 1, name: "Company A", growth: 25 },
      { id: 2, name: "Company B", growth: 22 },
    ]);
  });
});
