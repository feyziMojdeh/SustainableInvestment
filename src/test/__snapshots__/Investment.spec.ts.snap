// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Investment > match snapshot 1`] = `
"<div class="card">
  <tabs-stub value="0"></tabs-stub>
</div>"
`;
